library/ziti.c

```diff
@@ -992,11 +1033,31 @@ static void update_identity_data(ziti_identity_data *data, const ziti_error *err
     update_ctrl_status(ztx, FIELD_OR_ELSE(err, err, 0), FIELD_OR_NULL(err, message));
 }
 
+void update_session_data(ziti_session *session, const ziti_error *err, void *ctx) {
+    ziti_context ztx = ctx;
+
+    if (err != NULL) {
+        ZTX_LOG(ERROR, "error refreshing current api session immediately after auth query processing: err[%s]", ziti_errorstr(err->err));
+    }
+
+    if (session) {
+        ziti_set_api_session(ztx, session);
+        ZTX_LOG(VERBOSE, "api session after authentication query processing complete");
+    }
+}
+
 static void session_post_auth_query_cb(ziti_context ztx, int status, void *ctx) {
-    ziti_session *session = ztx->session;
+    ziti_session *session = ztx->api_session;
+
+    ziti_set_fully_authenticated(ztx);
```

Comment:
should this depend on `status`? what if authcode was not accepted?

Reply:
It should looking at the code. 

When a non-ZITI_OK status is provided what should we do? Right now I set it back to unauthenticated, but if we have auth queries that don't pause on user input, this will repeat on the session_refresh timer.